hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -153,6 +153,17 @@ public BasicOzoneClientAdapterImpl(String omHost, int omPort,
     objectStore = ozoneClient.getObjectStore();
     this.volume = objectStore.getVolume(volumeStr);
     this.bucket = volume.getBucket(bucketStr);
+
+    // Check if bucket layout is valid, OFS buckets cannot be in
+    // OBJECT_STORE layout
+    BucketLayout bucketLayout = bucket.getBucketLayout();
+    if (bucketLayout.equals(BucketLayout.OBJECT_STORE)) {
+      throw new IllegalArgumentException(bucketLayout + " does not support" +
```

Comment:
@JyotinderSingh Please modify the message with bucket name, something like:

  ```
    throw new IllegalArgumentException("Bucket: " + bucket.getName() +" is with layout: " + bucketLayout + ", which does not support" +
          " file system semantics. Bucket Layout must be " +
          BucketLayout.FILE_SYSTEM_OPTIMIZED + " or "
          + BucketLayout.LEGACY + ".");
```

Reply:
I will suggest to place this in a common class like FSUtils.java.